src/NewsService.Mappers/Models/NewsInfoMapper.cs

```diff
@@ -1,106 +1,41 @@
-    private readonly IRequestClient<IGetDepartmentRequest> _departmentRequestClient;
-    private readonly ILogger _logger;
+    private readonly IUserInfoMapper _userInfoMapper;
+    private readonly IDepartmentInfoMapper _departmentInfoMapper;
 
-    //TODO Remove to command
-    private string GetUserFullName(Guid userId)
+    public NewsInfoMapper(IUserInfoMapper userInfoMapper, IDepartmentInfoMapper departmentInfoMapper)
     {
-      string fullName = null;
-
-      try
-      {
-        object request = IGetUserDataRequest.CreateObj(userId);
-        Response<IOperationResult<IGetUserDataResponse>> response = _userRequestClient.GetResponse<IOperationResult<IGetUserDataResponse>>(request).Result;
-
-        if (!response.Message.IsSuccess)
-        {
-          _logger.LogWarning($"Can't found user. Reason: '{string.Join(',', response.Message.Errors)}'");
-        }
-        else
-        {
-          fullName = $"{response.Message.Body.FirstName} {response.Message.Body.LastName} {response.Message.Body.MiddleName}".Trim();
-        }
-      }
-      catch (Exception exception)
-      {
-        _logger.LogError(exception, "Exception on get user data request.");
-      }
-
-      return fullName;
-    }
-
-    private string GetDepartmentName(Guid departmentId)
-    {
-      string name = null;
-
-      try
-      {
-        object request = IGetDepartmentRequest.CreateObj(null, departmentId);
-        Response<IOperationResult<IGetDepartmentResponse>> response = _departmentRequestClient.GetResponse<IOperationResult<IGetDepartmentResponse>>(request).Result;
-
-        if (!response.Message.IsSuccess)
-        {
-          _logger.LogWarning($"Can't found department name. Reason: '{string.Join(',', response.Message.Errors)}'");
-        }
-        else
-        {
-          name = response.Message.Body.Name;
-        }
-      }
-      catch (Exception exception)
-      {
-        _logger.LogError(exception, "Exception on get department data request.");
-      }
-
-      return name;
+      _userInfoMapper = userInfoMapper;
+      _departmentInfoMapper = departmentInfoMapper;
     }
 
-    public NewsInfoMapper(
-      IRequestClient<IGetUserDataRequest> userClient,
-      IRequestClient<IGetDepartmentRequest> departmentClient,
-      ILogger<NewsInfoMapper> logger)
-    {
-      _userRequestClient = userClient;
-      _departmentRequestClient = departmentClient;
-      _logger = logger;
-    }
-
-    public NewsInfo Map(DbNews dbNews)
+    public NewsInfo Map(DbNews dbNews, List<DepartmentData> departments, List<UserData> authors)
     {
       if (dbNews == null)
       {
         return null;
       }
 
-      Department department = null;
-      if (dbNews.DepartmentId != null && dbNews.DepartmentId != Guid.Empty)
-      {
-        department = new Department { Id = (Guid)dbNews.DepartmentId, Name = GetDepartmentName((Guid)dbNews.DepartmentId) };
-      }
+      List<ImageInfo> avatarImage = new();
```

Comment:
i guess, Is it always will be null in 38 string. For what in here?

Reply:
done